[quant][graphmode] fused conv3d + relu

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36994 [quant][graphmode] Test refactor -- extract graph checker
* #36889 [quant][graphmode] linear_relu
* #36887 [quant][graphmode] torch.clamp
* **#36885 [quant][graphmode] fused conv3d + relu**

Differential Revision: [D21110359](https://our.internmc.facebook.com/intern/diff/D21110359)

Commit:
[quant][graphmode] fused conv3d + relu

[ghstack-poisoned]

Commit:
Update on "[quant][graphmode] fused conv3d + relu"


Differential Revision: [D21110359](https://our.internmc.facebook.com/intern/diff/D21110359)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1243,6 +1244,29 @@ def forward(self, x):
+                 torch.randint(0, 1, (1,), dtype=torch.long)) for _ in range(2)]
+        model = self._test_op_impl(M(), data, "quantized::conv3d_relu")
+
+        FileCheck().check_not("aten::conv3d") \
+                   .check_not("aten::relu") \
+                   .check_not("quantized::conv3d(") \
+                   .check_not("quantized::relu(") \
+                   .run(model.graph)
```
maybe we can integrate this to the `_test_op_impl` API as well, we can have a list of `check_not` ops

jerryzh168 replied:
I have some concern about whether calling `check_not` this way works or not, I know calling it separately definitely works, but I remember there is some problem when I chain them together

z-a-f replied:
> maybe we can integrate this to the `_test_op_impl` API as well, we can have a list of `check_not` ops

I like this idea. However, I think I should send it as a separate PR.

z-a-f replied:
> I have some concern about whether calling `check_not` this way works or not, I know calling it separately definitely works, but I remember there is some problem when I chain them together

I can set them up as checks on separate lines

z-a-f replied:
> I have some concern about whether calling `check_not` this way works or not, I know calling it separately definitely works, but I remember there is some problem when I chain them together

Can you elaborate what kind of problems you saw -- I want to check for it.

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1243,6 +1244,29 @@ def forward(self, x):
+                         " Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs"
+                         " with instruction set support avx2 or newer.")
+    def test_quantized_conv3d_relu(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv3d(1, 4, 2, 3).float()
+                self.relu = torch.nn.ReLU()
+
+            def forward(self, x):
+                return self.relu(self.conv(x))
```
could you add a test for functional relu as well?

Commit:
Update on "[quant][graphmode] fused conv3d + relu"


Differential Revision: [D21110359](https://our.internmc.facebook.com/intern/diff/D21110359)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmode] fused conv3d + relu"


Differential Revision: [D21110359](https://our.internmc.facebook.com/intern/diff/D21110359)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmode] fused conv3d + relu"


Differential Revision: [D21110359](https://our.internmc.facebook.com/intern/diff/D21110359)

[ghstack-poisoned]
